File: Temp/leetcode/editor/cn/[3174]清除数字.c
// 3174 清除数字
// https://leetcode.cn/problems/clear-digits/

//leetcode submit region begin(Prohibit modification and deletion)
char* clearDigits(char* s) {
    int top = 0; // 栈顶，指向接下来要填的位置
    for (int i = 0; s[i]; i++) {
        if (isdigit(s[i])) {
            top--; // 出栈
        } else {
            s[top++] = s[i]; // 入栈
        }
    }
    s[top] = '\0';
    return s;
}
//leetcode submit region end(Prohibit modification and deletion)

File: Temp/leetcode/editor/cn/[3174]清除数字.py
# 3174 清除数字
# https://leetcode.cn/problems/clear-digits/


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def clearDigits(self, s: str) -> str:
        st = []
        for c in s:
            if c.isdigit():
                st.pop()
            else:
                st.append(c)
        return "".join(st)
# leetcode submit region end(Prohibit modification and deletion)
